geode-management/src/main/java/org/apache/geode/management/configuration/ClassName.java

```diff
@@ -76,20 +80,31 @@ public ClassName(String className, String jsonInitProperties) {
     }
   }
 
+  /**
+   * @param className this class needs to have a no-arg constructor
+   * @param properties the initialization properties
+   *        that will be passed to {@link org.apache.geode.cache.Declarable#initialize}.
+   */
   public ClassName(String className, Properties properties) {
     this.className = className;
     this.initProperties = properties;
   }
 
+  /**
+   * @return the name of the class
+   */
   public String getClassName() {
     return className;
   }
 
+  /**
+   * @return the properties that will be used when an instance is created.
+   */
   public Properties getInitProperties() {
     return initProperties;
   }
 
-  public static boolean isClassNameValid(String fqcn) {
+  private static boolean isClassNameValid(String fqcn) {
```

Comment:
why we make this private and then repeat the logic in another place? Would it be better to make this public so that other people don't have to repeat the effort?

Reply:
The reason for making it private on this class is that this is an external class and our supported APIs do not need expose this functionality.
I will find a common "internal" class that can host this functionality.

Reply:
I added a ManagementHelper class that is internal and has isClassNameValid. It should be available on both the client and the server.